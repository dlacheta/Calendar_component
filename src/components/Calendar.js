import React from "react";
import Price from "./Price";
import Dates from "./Dates";
import PropTypes from "prop-types";

const Calendar = ({
  price,
  priceCountry,
  rating,
  ratingAmount,
  checkInVisible,
  checkOutVisible,
  toogleCheckIn,
  toogleCheckOut,
  formDate,
  nextMonth,
  previousMonth,
  availableDates,
  pickCheckIn,
  pickCheckOut,
  checkInDate,
  checkOutDate,
  onChangeCheckIn,
  onChangeCheckOut
}) => (
  <div className="container">
    <Price
      price={price}
      priceCountry={priceCountry}
      rating={rating}
      ratingAmount={ratingAmount}
    />
    <Dates
      checkInVisible={checkInVisible}
      checkOutVisible={checkOutVisible}
      toogleCheckIn={toogleCheckIn}
      toogleCheckOut={toogleCheckOut}
      formDate={formDate}
      nextMonth={nextMonth}
      previousMonth={previousMonth}
      availableDates={availableDates}
      pickCheckIn={pickCheckIn}
      pickCheckOut={pickCheckOut}
      checkInDate={checkInDate}
      checkOutDate={checkOutDate}
      onChangeCheckIn={onChangeCheckIn}
      onChangeCheckOut={onChangeCheckOut}
    />
    <button type="submit" className="search">
      Search
    </button>
  </div>
);

Calendar.propTypes = {
  price: PropTypes.number.isRequired,
  priceCountry: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  ratingAmount: PropTypes.number.isRequired,
  checkInVisible: PropTypes.bool.isRequired,
  checkOutVisible: PropTypes.bool.isRequired,
  toogleCheckIn: PropTypes.func.isRequired,
  toogleCheckOut: PropTypes.func.isRequired,
  formDate: PropTypes.object.isRequired,
  nextMonth: PropTypes.func.isRequired,
  previousMonth: PropTypes.func.isRequired,
  availableDates: PropTypes.arrayOf(PropTypes.string),
  pickCheckIn: PropTypes.func.isRequired,
  pickCheckOut: PropTypes.func.isRequired,
  checkInDate: PropTypes.string,
  checkOutDate: PropTypes.string,
  onChangeCheckIn: PropTypes.func.isRequired,
  onChangeCheckOut: PropTypes.func.isRequired
};

Calendar.defaultProps = {
  availableDates: [],
  checkInDate: "",
  checkOutDate: ""
};

export default Calendar;
