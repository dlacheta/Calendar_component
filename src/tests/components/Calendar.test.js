import React from "react";
import { shallow } from "enzyme";
import Calendar from "../../components/Calendar";

it("Calendar component renders correctly", () => {
  const toogleCheckIn = jest.fn();
  const toogleCheckOut = jest.fn();
  const nextMonth = jest.fn();
  const previousMonth = jest.fn();
  const pickCheckIn = jest.fn();
  const pickCheckOut = jest.fn();
  const onChangeCheckIn = jest.fn();
  const onChangeCheckOut = jest.fn();
  const price = 298;
  const priceCountry = "z≈Ç";
  const rating = 4;
  const ratingAmount = 123;
  const checkInVisible = false;
  const checkOutVisible = false;
  const checkInDate = "";
  const checkOutDate = "";
  const formDate = new Date();
  const availableDates = [
    "4/12/2018",
    "5/12/2018",
    "6/12/2018",
    "7/12/2018",
    "8/12/2018",
    "9/12/2018",
    "10/12/2018",
    "11/12/2018",
    "12/12/2018",
    "13/12/2018",
    "14/12/2018",
    "15/12/2018",
    "16/12/2018",
    "17/12/2018",
    "18/12/2018",
    "19/12/2018"
  ];
  const wrapper = shallow(
    <Calendar
      toogleCheckIn={toogleCheckIn}
      toogleCheckOut={toogleCheckOut}
      nextMonth={nextMonth}
      previousMonth={previousMonth}
      pickCheckIn={pickCheckIn}
      pickCheckOut={pickCheckOut}
      onChangeCheckIn={onChangeCheckIn}
      onChangeCheckOut={onChangeCheckOut}
      price={price}
      priceCountry={priceCountry}
      rating={rating}
      ratingAmount={ratingAmount}
      checkInVisible={checkInVisible}
      checkOutVisible={checkOutVisible}
      formDate={formDate}
      availableDates={availableDates}
      checkInDate={checkInDate}
      checkOutDate={checkOutDate}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
