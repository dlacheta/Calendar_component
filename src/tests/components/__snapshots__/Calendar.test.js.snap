// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calendar
    availableDates={
      Array [
        "4/12/2018",
        "5/12/2018",
        "6/12/2018",
        "7/12/2018",
        "8/12/2018",
        "9/12/2018",
        "10/12/2018",
        "11/12/2018",
        "12/12/2018",
        "13/12/2018",
        "14/12/2018",
        "15/12/2018",
        "16/12/2018",
        "17/12/2018",
        "18/12/2018",
        "19/12/2018",
      ]
    }
    checkInDate=""
    checkInVisible={false}
    checkOutDate=""
    checkOutVisible={false}
    formDate={2018-12-24T12:33:16.627Z}
    nextMonth={[MockFunction]}
    onChangeCheckIn={[MockFunction]}
    onChangeCheckOut={[MockFunction]}
    pickCheckIn={[MockFunction]}
    pickCheckOut={[MockFunction]}
    previousMonth={[MockFunction]}
    price={298}
    priceCountry="zł"
    rating={4}
    ratingAmount={123}
    toogleCheckIn={[MockFunction]}
    toogleCheckOut={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Price
          price={298}
          priceCountry="zł"
          rating={4}
          ratingAmount={123}
        />,
        <Dates
          availableDates={
            Array [
              "4/12/2018",
              "5/12/2018",
              "6/12/2018",
              "7/12/2018",
              "8/12/2018",
              "9/12/2018",
              "10/12/2018",
              "11/12/2018",
              "12/12/2018",
              "13/12/2018",
              "14/12/2018",
              "15/12/2018",
              "16/12/2018",
              "17/12/2018",
              "18/12/2018",
              "19/12/2018",
            ]
          }
          checkInDate=""
          checkInVisible={false}
          checkOutDate=""
          checkOutVisible={false}
          formDate={2018-12-24T12:33:16.627Z}
          nextMonth={[MockFunction]}
          onChangeCheckIn={[MockFunction]}
          onChangeCheckOut={[MockFunction]}
          pickCheckIn={[MockFunction]}
          pickCheckOut={[MockFunction]}
          previousMonth={[MockFunction]}
          toogleCheckIn={[MockFunction]}
          toogleCheckOut={[MockFunction]}
        />,
        <button
          className="search"
          type="submit"
        >
          Search
        </button>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "price": 298,
          "priceCountry": "zł",
          "rating": 4,
          "ratingAmount": 123,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "availableDates": Array [
            "4/12/2018",
            "5/12/2018",
            "6/12/2018",
            "7/12/2018",
            "8/12/2018",
            "9/12/2018",
            "10/12/2018",
            "11/12/2018",
            "12/12/2018",
            "13/12/2018",
            "14/12/2018",
            "15/12/2018",
            "16/12/2018",
            "17/12/2018",
            "18/12/2018",
            "19/12/2018",
          ],
          "checkInDate": "",
          "checkInVisible": false,
          "checkOutDate": "",
          "checkOutVisible": false,
          "formDate": 2018-12-24T12:33:16.627Z,
          "nextMonth": [MockFunction],
          "onChangeCheckIn": [MockFunction],
          "onChangeCheckOut": [MockFunction],
          "pickCheckIn": [MockFunction],
          "pickCheckOut": [MockFunction],
          "previousMonth": [MockFunction],
          "toogleCheckIn": [MockFunction],
          "toogleCheckOut": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
          "className": "search",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Price
            price={298}
            priceCountry="zł"
            rating={4}
            ratingAmount={123}
          />,
          <Dates
            availableDates={
              Array [
                "4/12/2018",
                "5/12/2018",
                "6/12/2018",
                "7/12/2018",
                "8/12/2018",
                "9/12/2018",
                "10/12/2018",
                "11/12/2018",
                "12/12/2018",
                "13/12/2018",
                "14/12/2018",
                "15/12/2018",
                "16/12/2018",
                "17/12/2018",
                "18/12/2018",
                "19/12/2018",
              ]
            }
            checkInDate=""
            checkInVisible={false}
            checkOutDate=""
            checkOutVisible={false}
            formDate={2018-12-24T12:33:16.627Z}
            nextMonth={[MockFunction]}
            onChangeCheckIn={[MockFunction]}
            onChangeCheckOut={[MockFunction]}
            pickCheckIn={[MockFunction]}
            pickCheckOut={[MockFunction]}
            previousMonth={[MockFunction]}
            toogleCheckIn={[MockFunction]}
            toogleCheckOut={[MockFunction]}
          />,
          <button
            className="search"
            type="submit"
          >
            Search
          </button>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "price": 298,
            "priceCountry": "zł",
            "rating": 4,
            "ratingAmount": 123,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "availableDates": Array [
              "4/12/2018",
              "5/12/2018",
              "6/12/2018",
              "7/12/2018",
              "8/12/2018",
              "9/12/2018",
              "10/12/2018",
              "11/12/2018",
              "12/12/2018",
              "13/12/2018",
              "14/12/2018",
              "15/12/2018",
              "16/12/2018",
              "17/12/2018",
              "18/12/2018",
              "19/12/2018",
            ],
            "checkInDate": "",
            "checkInVisible": false,
            "checkOutDate": "",
            "checkOutVisible": false,
            "formDate": 2018-12-24T12:33:16.627Z,
            "nextMonth": [MockFunction],
            "onChangeCheckIn": [MockFunction],
            "onChangeCheckOut": [MockFunction],
            "pickCheckIn": [MockFunction],
            "pickCheckOut": [MockFunction],
            "previousMonth": [MockFunction],
            "toogleCheckIn": [MockFunction],
            "toogleCheckOut": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "search",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
